test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/HtmlHelperDropDownListExtensionsTest.cs

```diff
@@ -0,0 +1,232 @@
+                "<option value=\"HtmlEncode[[3]]\">HtmlEncode[[Three]]</option>" + Environment.NewLine +
+                "</select>";
+            var metadataProvider = new EmptyModelMetadataProvider();
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(new ViewDataDictionary<TestModel>(metadataProvider));
+            helper.ViewContext.ClientValidationEnabled = false;
+            helper.ViewData.Model = new TestModel { Property1 = BasicSelectList };
+
+            // Act
+            var dropDownListResult = helper.DropDownList("Property1", "--select--");
+
+            // Assert
+            Assert.Equal(expectedHtml, HtmlContentUtilities.HtmlContentToString(dropDownListResult));
+        }
+
+        [Fact]
+        public void DropDownList_UsesSpecifiedExpressionAndSelectList()
+        {
+            // Arrange
+            var expectedHtml = "<select id=\"HtmlEncode[[Property1]]\" name=\"HtmlEncode[[Property1]]\">" +
+                "<option value=\"HtmlEncode[[4]]\">HtmlEncode[[Four]]</option>" + Environment.NewLine +
+                "<option value=\"HtmlEncode[[5]]\">HtmlEncode[[Five]]</option>" + Environment.NewLine +
+                "</select>";
+            var metadataProvider = new EmptyModelMetadataProvider();
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(new ViewDataDictionary<TestModel>(metadataProvider));
+            helper.ViewContext.ClientValidationEnabled = false;
+            helper.ViewData.Model = new TestModel { Property1 = BasicSelectList };
+
+            // Act
+            var dropDownListResult = helper.DropDownList("Property1", new List<SelectListItem>
+            {
+                new SelectListItem { Text = "Four", Value = "4" },
+                new SelectListItem { Text = "Five", Value = "5" },
+            });
+
+            // Assert
+            Assert.Equal(expectedHtml, HtmlContentUtilities.HtmlContentToString(dropDownListResult));
+        }
+
+        [Fact]
+        public void DropDownListFor_UsesSpecifiedExpressionAndSelectList()
+        {
+            // Arrange
+            var expectedHtml = "<select id=\"HtmlEncode[[Property1]]\" name=\"HtmlEncode[[Property1]]\">" +
+                "<option value=\"HtmlEncode[[4]]\">HtmlEncode[[Four]]</option>" + Environment.NewLine +
+                "<option value=\"HtmlEncode[[5]]\">HtmlEncode[[Five]]</option>" + Environment.NewLine +
+                "</select>";
+            var metadataProvider = new EmptyModelMetadataProvider();
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(new ViewDataDictionary<TestModel>(metadataProvider));
+            helper.ViewContext.ClientValidationEnabled = false;
+            helper.ViewData.Model = new TestModel { Property1 = BasicSelectList };
+
+            // Act
+            var dropDownListForResult = helper.DropDownListFor(m => m.Property1, new List<SelectListItem>
```

Comment:
In this case it's fine to place the select list in VDD directly: `helper.ViewData["Property1"] = BasicSelectList;` But the passed expression should _never_ refer to a select list. Give this property a real value in the model.